lib/rules/no-irregular-whitespace/index.js

```diff
@@ -0,0 +1,136 @@
+// @ts-nocheck
+
+'use strict';
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'no-irregular-whitespace';
+const messages = ruleMessages(ruleName, {
+	unexpected: 'Unexpected irregular whitespace',
+});
+
+const IRREGULAR_WHITESPACES = [
+	'\u000B', // Line Tabulation (\v) - <VT>
+	'\u000C', // Form Feed (\f) - <FF>
+	'\u00A0', // No-Break Space - <NBSP>
+	'\u0085', // Next Line
+	'\u1680', // Ogham Space Mark
+	'\u180E', // Mongolian Vowel Separator - <MVS>
+	'\uFEFF', // Zero Width No-Break Space - <BOM>
+	'\u2000', // En Quad
+	'\u2001', // Em Quad
+	'\u2002', // En Space - <ENSP>
+	'\u2003', // Em Space - <EMSP>
+	'\u2004', // Tree-Per-Em
+	'\u2005', // Four-Per-Em
+	'\u2006', // Six-Per-Em
+	'\u2007', // Figure Space
+	'\u2008', // Punctuation Space - <PUNCSP>
+	'\u2009', // Thin Space
+	'\u200A', // Hair Space
+	'\u200B', // Zero Width Space - <ZWSP>
+	'\u2028', // Line Separator
+	'\u2029', // Paragraph Separator
+	'\u202F', // Narrow No-Break Space
+	'\u205F', // Medium Mathematical Space
+	'\u3000', // Ideographic Space
+];
+
+const generateInvalidWhitespaceValidator = (ignoreArray = []) => {
+	const pattern = new RegExp(
+		`(${IRREGULAR_WHITESPACES.filter((ws) => !ignoreArray.includes(ws)).join('|')})`,
+	);
+
+	return (str) => typeof str === 'string' && pattern.exec(str);
+};
```

Comment:
[suggestion] Perhaps, it seems we can simplify `generateInvalidWhitespaceValidator()` as below:

```suggestion
const IRREGULAR_WHITESPACES_PATTERN = new RegExp(`([${IRREGULAR_WHITESPACES.join('')}])`);

const generateInvalidWhitespaceValidator = () => {
	return (str) => typeof str === 'string' && IRREGULAR_WHITESPACES_PATTERN.exec(str);
};
```

Reply:
I agree, thanks! 👍